monkey/monkey_island/cc/island_event_handlers/agent_heartbeat_handler.py

```diff
@@ -0,0 +1,40 @@
+from datetime import datetime, timezone
+from typing import Dict
+
+from common import AgentHeartbeat
+from common.common_consts import HEARTBEAT_INTERVAL
+from common.types import AgentID
+from monkey_island.cc.repositories import IAgentRepository
+
+
+class AgentHeartbeatHandler:
+    """
+    Tracks Agents' latest heartbeats and handles related functionality
+    """
+
+    def __init__(self, agent_repository: IAgentRepository):
+        self._agent_repository = agent_repository
+        self._latest_heartbeats: Dict[AgentID, datetime] = {}
+
+    def update_latest_heartbeat_of_agent(self, agent_id: AgentID, heartbeat: AgentHeartbeat):
+        self._latest_heartbeats[agent_id] = heartbeat.timestamp
+
+    def check_status_of_agents_from_latest_heartbeats(self):
```

Comment:
`check_...` is a bad name for this. It does more than just "check", it updates the repository.